### PR TITLE
FEATURE:MSF-19830 - Fix sonarqube errors

### PR DESCRIPTION
**Pull request by sangtm**

### CODE REVIEW

File: lib/gooddata/helpers/global_helpers.rb [hunk 1]
```diff
@@ -222,7 +225,7 @@ def to_boolean(param)
       # encrypts data with the given key. returns a binary data with the
       # unhashed random iv in the first 16 bytes
       def encrypt(data, key)
-        cipher = OpenSSL::Cipher::Cipher.new('aes-256-cbc')
+        cipher = OpenSSL::Cipher::Cipher.new(DEFAULT_CIPHER)
```
Comment (danh-ung) [inline lib/gooddata/helpers/global_helpers.rb:228]: Better name: AES_256_CBC_CIPHER

File: lib/gooddata/rest/connection.rb [hunk 2]
```diff
@@ -713,11 +716,11 @@ def webdav_dir_exists?(url)
       def fix_server_url(server)
         server = server.chomp('/')
         if server.start_with? 'http://'
-          server = server.sub 'http://', 'https://'
+          server = server.sub 'http://', PROTOCOL
```
Comment (danh-ung) [inline lib/gooddata/rest/connection.rb:719]: We should define 2 constants: HTTP_PROTOCOL and HTTPS_PROTOCOL

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-oss-check/559/ : SUCCESS in 7s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-lcm-build-pipeline/556/ : FAILURE in 8m 01s

**Comment by sangtm** [comment]
extended check sonar

**Comment by yenkins** [comment]
### Sonar scan result
 - 1 BUG
 - 18 CODE_SMELL

More detail, see in https://sonarqube-dev.intgdc.com/dashboard?id=gooddata-ruby-check-extended-sonar-PR1725

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-sonarqube-scan-zuul-el7/16/ : FAILURE in 1m 36s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-oss-check/560/ : SUCCESS in 7s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-oss-check/561/ : SUCCESS in 7s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-lcm-build-pipeline/557/ : SUCCESS in 23m 00s

**Comment by sangtm** [comment]
extended check sonar

**Comment by yenkins** [comment]
### Sonar scan result
 - No new bug found. Congratulation!

More detail, see in https://sonarqube-dev.intgdc.com/dashboard?id=gooddata-ruby-check-extended-sonar-PR1725

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-sonarqube-scan-zuul-el7/17/ : SUCCESS in 1m 52s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-oss-check/563/ : SUCCESS in 6s

**Comment by gdgate** [comment]
Build failed (gate pipeline).

- gooddata-ruby-sonarqube-scan-zuul-el7 : SKIPPED (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-merge-compliance-check/497/ : SUCCESS in 17s
- https://checklist.intgdc.com/job/lcm/job/gooddata-ruby-travis-check/148/ : FAILURE in 6s
- gooddata-ruby-fossa-scanning-zuul-el7 : SKIPPED (non-voting)

**Comment by danh-ung** [comment]
ok to test

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-oss-check/565/ : SUCCESS in 1m 13s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-lcm-build-pipeline/560/ : SUCCESS in 20m 33s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-oss-check/566/ : SUCCESS in 8s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-lcm-build-pipeline/561/ : SUCCESS in 20m 26s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-oss-check/568/ : SUCCESS in 6s
- https://jenkins-ii.intgdc.com/job/gooddata-ruby/job/gooddata-ruby-lcm-build-pipeline/563/ : SUCCESS in 20m 53s

**Comment by yenkins** [comment]
### Sonar scan result
 - No new bug found. Congratulation!

More detail, see in https://sonarqube-dev.intgdc.com/dashboard?id=gooddata-ruby-gate-PR1725

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-sonarqube-scan-zuul-el7/19/ : SUCCESS in 1m 31s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-merge-compliance-check/498/ : SUCCESS in 32s
- https://checklist.intgdc.com/job/lcm/job/gooddata-ruby-travis-check/149/ : SUCCESS in 54s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-fossa-scanning-zuul-el7/58/ : FAILURE in 5m 35s (non-voting)

**Pull request merged**